File: src/Engine/DumbAIPlayer.cpp
#include "DumbAIPlayer.h"

#include <chrono>
#include <random>

namespace BeitaGo {
	DumbAIPlayer::DumbAIPlayer(Engine& engine, Color color) : AIPlayer(engine, color) {}

	Grid2 DumbAIPlayer::MakeDecision() const {
		auto validMoves = GetValidMoves();
		if (validMoves.empty()) {
			return PASS;
		} else {
			std::mt19937 rand(static_cast<unsigned int>(std::chrono::system_clock::now().time_since_epoch().count()));
			std::uniform_int_distribution<int> distribution(0, validMoves.size() - 1);
			return validMoves[distribution(rand)];
		}
	}
}

File: srcpy/engine/dumbaiplayer.py
import random
from .aiplayer import AIPlayer
from .constants import Color, PASS
from .grid2 import Grid2


class DumbAIPlayer(AIPlayer):
	from .engine import Engine

	def __init__(self, engine: Engine, color: Color):
		super().__init__(engine, color)

	def make_decision(self) -> Grid2:
		valid_moves = self.get_valid_moves()
		if len(valid_moves) == 0:
			return PASS
		else:
			return valid_moves[random.randint(0, len(valid_moves) - 1)]
